docs-internal/features/features.yaml

```diff
@@ -0,0 +1,87 @@
+# This is the configuration of features for Helidon
+# This file is part of repository in a new module
+# Output of the module is going to be
+# a feature-db.json (installed to maven central) containing
+# the full feature configuration, including coordinates
+
+features:
+  # The "name" is significant
+  #  - must be the same in @Feature annotation on module-info of feature module
+  #  - must be used by CLI
+  - name: "Security"
+    plugins:
+      # Type of plugin (used in CLI to list impls)
+      # This is a named list (feature is "Security/Providers/ABAC")
+      - name: "Providers"
+        service: "io.helidon.security.spi.SecurityProviderService"
+        # explicit list if sub-features are needed
+        list:
+          - name: "ABAC"
+            plugins:
+              - name: "Validators"
+                # each feature implementing this service is valid here
+                service: "io.helidon.security.providers.abac.spi.AbacValidatorService"
+  - name: "Tracing"
+    implementations:
+      service: "io.helidon.tracing.spi.TracerProvider"
+  - name: "DbClient"
+    drivers:
+      - name: "DbClient Type"
+        service: "io.helidon.dbclient.spi.DbClientProvider"
+        # explicitly list JDBC, as it requires sub-drivers (mongo is loaded through service loader)
+        list:
+          - name: "JDBC"
+            drivers:
+              type: "external"
+              service: "java.sql.Driver"
+              # maybe better for external:
+              category: "JDBC Driver"
+  - name: "JPA"
+    implementations:
+      list:
+        - name: "Hibernate"
+        - name: "EclipseLink"
+    drivers:
+      - name: "Database Driver"
+        type: "external"
+        service: "java.sql.Driver"
+        # maybe better for external:
+        category: "JDBC Driver"
+  - name: "Config"
+    plugins:
+      - name: "Sources"
+        service: "io.helidon.config.spi.ConfigSourceProvider"
+      - name: "Parsers"
+        service: "io.helidon.config.spi.ConfigParser"
+
+
+integrations:
+  # name is the feature path (may have more than one integration for same feature)
+  - group:
+    name: "Security"
+    integrations:
+      - name: "Security/Integration/WebServer"
+        when: ["Security","WebServer"]
```

Comment:
"Security" is not a features, it's some sort of path prefix right ? I guess It means all features with a path that starts with "Security".

Why not use a wildcard expression here, e.g. `security/*`

Also since we have a path, we should have some rules for the segments, e.g. no space etc. Right now we are using camel case for the paths which feels wrong. We should perhaps define an ID which composes the path, a name which is human readable (e.g. can be camel cased, can have spaces etc).

The ID is meant to be used for referring to features, it's defined in paths like here, it's declared in the annotation, and might also be referenced in the archetype stuff.